src/context/cart-context.js

```diff
@@ -1,28 +1,33 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useContext, useState, useReducer, useEffect } from "react";
 import axios from "axios"
+import { cartReducer } from "../reducer";
+import {token} from "../utils/token"
 
 const CartContext = createContext();
 
-function CartProvider({children}) {
+export function CartProvider({children}) {
 
     const [myCart, setMyCart] = useState([])
 
     const addToCart = async (token, product) => {
-      try {
-        const res = await axios.post('/api/user/cart', {product}, {
-          headers: {
-            authorization: token
-          }
-        })
-        setMyCart(res.data.cart)
-      }
-  
-      catch(err) {
-        console.log(err)
+      if(product !== null) {
+        try {
+          const res = await axios.post('/api/user/cart', {product}, {
+            headers: {
+              authorization: token
+            }
+          })
+          setMyCart(res.data.cart)
+        }
+    
+        catch(err) {
+          console.log(err)
+        }
       }
     }
 
-    const deleteFromCart = async (token, productId) => {
+    const deleteFromCart = async (token, productId, deleteProduct) => {
+      if(productId !== null && deleteProduct===true) {
```

Comment:
it deleteproduct is a boolean value no need to add ===true just if( deleteProduct) will do.